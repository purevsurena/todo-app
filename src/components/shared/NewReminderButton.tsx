import React from "react";
import { View, StyleSheet } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { ThemedText } from "@/components/shared/ThemedText";
import Click from "@/components/layouts/Click";

interface NewReminderButtonProps {
  onPress: () => void;
  color: string;
}

const NewReminderButton: React.FC<NewReminderButtonProps> = ({
  onPress,
  color,
}) => {
  return (
    <Click style={styles.container} onPress={onPress}>
      <View>
        <MaterialCommunityIcons name="plus-circle" size={26} color={color} />
      </View>
      <ThemedText style={styles.text} type="default">
        New Reminder
      </ThemedText>
    </Click>
  );
};

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    bottom: 64,
    right: 26,
    flexDirection: "row",
    justifyContent: "flex-end",
    alignItems: "center",
  },
  text: {
    marginLeft: 4,
    fontWeight: "500",
    fontSize: 16,
  },
});

export default NewReminderButton;
