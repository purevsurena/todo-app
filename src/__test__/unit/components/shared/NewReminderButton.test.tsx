import React from "react";
import { render } from "@testing-library/react-native";
import NewReminderButton from "@/components/shared/NewReminderButton";

describe("NewReminderButton Component", () => {
  it("matches the snapshot", () => {
    const { toJSON } = render(
      <NewReminderButton onPress={() => {}} color="blue" />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
