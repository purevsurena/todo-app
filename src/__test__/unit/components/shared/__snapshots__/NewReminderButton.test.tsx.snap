// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewReminderButton Component matches the snapshot 1`] = `
[
  <View>
    <Text />
  </View>,
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 16,
          "fontWeight": "500",
          "marginLeft": 4,
        },
      ]
    }
  >
    New Reminder
  </Text>,
]
`;
